// android-emulator-webrtc/dist/components/emulator/net/emulator_status.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _empty_pb = require("google-protobuf/google/protobuf/empty_pb");
var _emulator_web_client = require("../../../proto/emulator_web_client");
/*
 * Copyright 2020 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License")
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/**
 * Gets the status of the emulator, parsing the hardware config into something
 * easy to digest.
 *
 * @export
 * @class EmulatorStatus
 */
var EmulatorStatus = /*#__PURE__*/(0, _createClass2["default"])(
/**
 * Creates an EmulatorStatus object that can retrieve the status of the running emulator.
 *
 * @param {object} uriOrEmulator An emulator controller service, or a URI to a gRPC endpoint.
 * @param {object} auth The authentication service to use, or null for no authentication.
 *
 *  The authentication service should implement the following methods:
 * - `authHeader()` which must return a set of headers that should be send along with a request.
 * - `unauthorized()` a function that gets called when a 401 was received.
 */
function EmulatorStatus(uriOrEmulator, auth) {
  var _this = this;
  (0, _classCallCheck2["default"])(this, EmulatorStatus);
  /**
   * Gets the cached status.
   *
   * @memberof EmulatorStatus
   */
  (0, _defineProperty2["default"])(this, "getStatus", function () {
    return _this.status;
  });
  /**
   * Retrieves the current status from the emulator.
   *
   * @param  {Callback} fnNotify when the status is available, returns the retrieved status.
   * @param  {boolean} cache True if the cache can be used.
   * @memberof EmulatorStatus
   */
  (0, _defineProperty2["default"])(this, "updateStatus", function (fnNotify, cache) {
    var request = new _empty_pb.Empty();
    if (cache && _this.status) {
      fnNotify(_this.status);
      return _this.status;
    }
    _this.emulator.getStatus(request, {}, function (err, response) {
      var hwConfig = {};
      var entryList = response.getHardwareconfig().getEntryList();
      for (var i = 0; i < entryList.length; i++) {
        var key = entryList[i].getKey();
        var val = entryList[i].getValue();
        hwConfig[key] = val;
      }
      var vmConfig = response.getVmconfig();
      _this.status = {
        version: response.getVersion(),
        uptime: response.getUptime(),
        booted: response.getBooted(),
        hardwareConfig: hwConfig,
        vmConfig: {
          hypervisorType: vmConfig.getHypervisortype(),
          numberOfCpuCores: vmConfig.getNumberofcpucores(),
          ramSizeBytes: vmConfig.getRamsizebytes()
        }
      };
      fnNotify(_this.status);
    });
  });
  if (uriOrEmulator instanceof _emulator_web_client.EmulatorControllerService) {
    this.emulator = uriOrEmulator;
  } else {
    this.emulator = new _emulator_web_client.EmulatorControllerService(uriOrEmulator, auth);
  }
  this.status = null;
});
var _default = EmulatorStatus;
exports["default"] = _default;
