// android-emulator-webrtc/dist/components/emulator/net/jsep_protocol_driver.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));
var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _events = require("events");
var _empty_pb = require("google-protobuf/google/protobuf/empty_pb");
/*
 * Copyright 2019 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License")
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/**
 * This drives the jsep protocol with the emulator, and can be used to
 * send key/mouse/touch events to the emulator. Events will be send
 * over the data channel if open, otherwise they will be send via the
 * grpc endpoint.
 *
 *  The jsep protocol is described here:
 * https://rtcweb-wg.github.io/jsep/.
 *
 *  This class can fire two events:
 *
 * - `connected` when the stream has become available.
 * - `disconnected` when the stream broke down, or when we failed to establish a connection
 *
 * You usually want to start the stream after instantiating this object. Do not forget to
 * disconnect once you are finished to terminate the message pump.
 *
 *
 * @example
 *  jsep = new JsepProtocolDriver(emulator, s => { video.srcObject = s; video.play() });
 *  jsep.startStream();
 *
 * @export
 * @class JsepProtocol
 */
var JsepProtocol = /*#__PURE__*/function () {
  /**
   * Creates an instance of JsepProtocol.
   * @param {EmulatorService} emulator Service used to make the gRPC calls
   * @param {RtcService} rtc Service used to open up the rtc calls.
   * @param {boolean} poll True if we should use polling
   * @param {callback} onConnect optional callback that is invoked when a stream is available
   * @param {callback} onDisconnect optional callback that is invoked when the stream is closed.
   * @memberof JsepProtocol
   */
  function JsepProtocol(emulator, rtc, poll, onConnect, onDisconnect) {
    var _this = this;
    (0, _classCallCheck2["default"])(this, JsepProtocol);
    (0, _defineProperty2["default"])(this, "on", function (name, fn) {
      _this.events.on(name, fn);
    });
    /**
     * Disconnects the stream. This will stop the message pump as well.
     *
     * @memberof JsepProtocol
     */
    (0, _defineProperty2["default"])(this, "disconnect", function () {
      _this.connected = false;
      if (_this.peerConnection) _this.peerConnection.close();
      if (_this.stream) {
        _this.stream.cancel();
        _this.stream = null;
      }
      _this.active = false;
      _this.old_emu_patch = {
        candidates: [],
        sdp: null,
        haveOffer: false,
        answer: false
      };
      _this.events.emit("disconnected", _this);
    });
    /**
     * Initiates the JSEP protocol.
     *
     * @memberof JsepProtocol
     */
    (0, _defineProperty2["default"])(this, "startStream", function () {
      var self = _this;
      _this.connected = false;
      _this.peerConnection = null;
      _this.active = true;
      _this.old_emu_patch = {
        candidates: [],
        sdp: null,
        haveOffer: false,
        answer: false
      };
      var request = new _empty_pb.Empty();
      _this.rtc.requestRtcStream(request, {}, function (err, response) {
        if (err) {
          console.error("Failed to configure rtc stream: " + JSON.stringify(err));
          _this.disconnect();
          return;
        }

        // Configure
        self.guid = response;
        self.connected = true;
        if (!_this.poll) {
          // Streaming envoy based.
          self._streamJsepMessage();
        } else {
          // Poll pump messages, go/envoy based proxy.
          console.debug("Polling jsep messages.");
          self._receiveJsepMessage();
        }
      });
    });
    (0, _defineProperty2["default"])(this, "cleanup", function () {
      _this.disconnect();
      if (_this.peerConnection) {
        _this.peerConnection.removeEventListener("track", _this._handlePeerConnectionTrack);
        _this.peerConnection.removeEventListener("icecandidate", _this._handlePeerIceCandidate);
        _this.peerConnection = null;
      }
      _this.event_forwarders = {};
    });
    (0, _defineProperty2["default"])(this, "_handlePeerConnectionTrack", function (e) {
      _this.events.emit("connected", e.track);
    });
    (0, _defineProperty2["default"])(this, "_handlePeerConnectionStateChange", function (e) {
      switch (_this.peerConnection.connectionState) {
        case "disconnected":
        // At least one of the ICE transports for the connection is in the "disconnected" state
        // and none of the other transports are in the state "failed", "connecting",
        // or "checking".
        case "failed":
        // One or more of the ICE transports on the connection is in the "failed" state.
        case "closed":
          //The RTCPeerConnection is closed.
          _this.disconnect();
      }
    });
    (0, _defineProperty2["default"])(this, "_handlePeerIceCandidate", function (e) {
      if (e.candidate === null) return;
      _this._sendJsep({
        candidate: e.candidate
      });
    });
    (0, _defineProperty2["default"])(this, "_handleDataChannel", function (e) {
      var channel = e.channel;
      _this.event_forwarders[channel.label] = channel;
    });
    (0, _defineProperty2["default"])(this, "_handleStart", function (signal) {
      _this.peerConnection = new RTCPeerConnection(signal.start);
      _this.peerConnection.ontrack = _this._handlePeerConnectionTrack;
      _this.peerConnection.onicecandidate = _this._handlePeerIceCandidate;
      _this.peerConnection.onconnectionstatechange = _this._handlePeerConnectionStateChange;
      _this.peerConnection.ondatachannel = _this._handleDataChannel;
      _this.peerConnection.onsignalingstatechange = _this._handlePeerState;
    });
    (0, _defineProperty2["default"])(this, "_handlePeerState", function (event) {
      if (!_this.peerConnection) {
        console.log("Peerconnection no longer available, ignoring signal state.");
      }
      switch (_this.peerConnection.signalingState) {
        case "have-remote-offer":
          _this.old_emu_patch.haveOffer = true;
          while (_this.old_emu_patch.candidates.length > 0) {
            _this._handleCandidate(_this.old_emu_patch.candidates.shift());
          }
          break;
      }
    });
    (0, _defineProperty2["default"])(this, "_handleSDP", /*#__PURE__*/function () {
      var _ref = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee(signal) {
        var answer;
        return _regenerator["default"].wrap(function _callee$(_context) {
          while (1) switch (_context.prev = _context.next) {
            case 0:
              // We should not call this more than once..
              _this.old_emu_patch.sdp = null;
              _this.peerConnection.setRemoteDescription(new RTCSessionDescription(signal));
              _context.next = 4;
              return _this.peerConnection.createAnswer();
            case 4:
              answer = _context.sent;
              if (answer) {
                // Older emulators cannot handle multiple answers, so make sure we do not send one
                // again.
                if (!_this.old_emu_patch.answer) {
                  _this.old_emu_patch.answer = true;
                  _this.peerConnection.setLocalDescription(answer);
                  _this._sendJsep({
                    sdp: answer
                  });
                }
              } else {
                _this.disconnect();
              }
            case 6:
            case "end":
              return _context.stop();
          }
        }, _callee);
      }));
      return function (_x) {
        return _ref.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])(this, "_handleCandidate", function (signal) {
      _this.peerConnection.addIceCandidate(new RTCIceCandidate(signal));
    });
    (0, _defineProperty2["default"])(this, "_handleSignal", function (signal) {
      if (signal.start) {
        _this._handleStart(signal);
      }
      if (signal.bye) {
        _this._handleBye();
      }
      if (signal.sdp && !_this.old_emu_patch.sdp) {
        _this.old_emu_patch.sdp = signal;
      }
      if (signal.candidate) {
        _this.old_emu_patch.candidates.push(signal);
      }
      if (!!_this.peerConnection) {
        // We have created a peer connection..
        if (_this.old_emu_patch.sdp) {
          _this._handleSDP(_this.old_emu_patch.sdp);
        }
        if (_this.old_emu_patch.haveOffer) {
          // We have a remote peer, add the candidates in.
          while (_this.old_emu_patch.candidates.length > 0) {
            _this._handleCandidate(_this.old_emu_patch.candidates.shift());
          }
        }
      }
    });
    (0, _defineProperty2["default"])(this, "_handleJsepMessage", function (message) {
      try {
        var signal = JSON.parse(message);
        _this._handleSignal(signal);
      } catch (e) {
        console.error("Failed to handle message: [" + message + "], due to: " + JSON.stringify(e));
      }
    });
    (0, _defineProperty2["default"])(this, "_handleBye", function () {
      if (_this.connected) {
        _this.disconnect();
      }
    });
    (0, _defineProperty2["default"])(this, "_sendJsep", function (jsonObject) {
      /* eslint-disable */
      var request = new proto.android.emulation.control.JsepMsg();
      request.setId(_this.guid);
      request.setMessage(JSON.stringify(jsonObject));
      _this.rtc.sendJsepMessage(request);
    });
    (0, _defineProperty2["default"])(this, "_streamJsepMessage", function () {
      if (!_this.connected) return;
      var self = _this;
      _this.stream = _this.rtc.receiveJsepMessages(_this.guid, {});
      _this.stream.on("data", function (response) {
        var msg = response.getMessage();
        self._handleJsepMessage(msg);
      });
      _this.stream.on("error", function (e) {
        console.log("Jsep message stream error:", e);
        if (self.connected) {
          console.log("Attempting to reconnect to jsep message stream.");
          self._streamJsepMessage();
        }
      });
      _this.stream.on("end", function (e) {
        if (self.connected) {
          console.log("Stream end while still connected.. Reconnecting");
          self._streamJsepMessage();
        }
      });
    });
    // This function is a fallback for v1 (go based proxy), that does not support streaming.
    (0, _defineProperty2["default"])(this, "_receiveJsepMessage", function () {
      if (!_this.connected) return;
      var self = _this;

      // This is a blocking call, that will return as soon as a series
      // of messages have been made available, or if we reach a timeout
      _this.rtc.receiveJsepMessage(_this.guid, {}, function (err, response) {
        if (err) {
          console.error("Failed to receive jsep message, disconnecting: " + JSON.stringify(err));
          _this.disconnect();
        }
        var msg = response.getMessage();
        // Handle only if we received a useful message.
        // it is possible to get nothing if the server decides
        // to kick us out.
        if (msg) {
          self._handleJsepMessage(response.getMessage());
        }

        // And pump messages. Note we must continue the message pump as we
        // can receive new ICE candidates at any point in time.
        if (self.active) {
          self._receiveJsepMessage();
        }
      });
    });
    this.emulator = emulator;
    this.rtc = rtc;
    this.events = new _events.EventEmitter();

    // Workaround for older emulators that send messages out of order
    // and do not handle all answers properly.
    this.old_emu_patch = {
      candidates: [],
      sdp: null,
      haveOffer: false,
      answer: false
    };
    this.poll = poll;
    this.guid = null;
    this.stream = null;
    this.event_forwarders = {};
    if (onConnect) this.events.on("connected", onConnect);
    if (onDisconnect) this.events.on("disconnected", onDisconnect);
  }
  (0, _createClass2["default"])(JsepProtocol, [{
    key: "send",
    value: function send(label, msg) {
      var bytes = msg.serializeBinary();
      var forwarder = this.event_forwarders[label];
      console.log("Send " + label + " " + JSON.stringify(msg.toObject()));
      // Send via data channel/gRPC bridge.
      if (this.connected && forwarder && forwarder.readyState == "open") {
        this.event_forwarders[label].send(bytes);
      } else {
        // Fallback to using the gRPC protocol
        switch (label) {
          case "mouse":
            this.emulator.sendMouse(msg);
            break;
          case "keyboard":
            this.emulator.sendKey(msg);
            break;
          case "touch":
            this.emulator.sendTouch(msg);
            break;
        }
      }
    }
  }]);
  return JsepProtocol;
}();
exports["default"] = JsepProtocol;
