// android-emulator-webrtc/dist/components/emulator/net/logcat.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _events = require("events");
require("../../../proto/emulator_controller_pb");
var _emulator_web_client = require("../../../proto/emulator_web_client");
/*
 * Copyright 2019 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License")
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/**
 * Observe the logcat stream from the emulator.
 *
 * Streaming is done by either polling the emulator endpoint or making a streaming call.
 *
 * It will send out the following events:
 *
 * - `start` whenever the start method was called.
 * - `data` whenever new data became available.
 * - `end` whenever the stream is finished, either because it was stopped, or due to an error.
 */
var Logcat = /*#__PURE__*/(0, _createClass2["default"])(
/**
 * Creates a logcat stream.
 *
 *  The authentication service should implement the following methods:
 * - `authHeader()` which must return a set of headers that should be send along with a request.
 * - `unauthorized()` a function that gets called when a 401 was received.
 *
 * @constructor
 * @param {object} uriOrEmulator
 * @param {object} auth
 */
function Logcat(uriOrEmulator, auth) {
  var _this = this;
  (0, _classCallCheck2["default"])(this, Logcat);
  /**
   * Register a listener.
   *
   * @param {string} name Name of the event.
   * @param  {Callback} fn Function to notify on the given event.
   * @memberof Logcat
   */
  (0, _defineProperty2["default"])(this, "on", function (name, fn) {
    _this.events.on(name, fn);
  });
  /**
   * Removes a listener.
   *
   * @param {string} name Name of the event.
   * @param  {Callback} fn Function to notify on the given event.
   * @memberof Logcat
   */
  (0, _defineProperty2["default"])(this, "off", function (name, fn) {
    _this.events.off(name, fn);
  });
  /**
   * Cancel the currently active logcat stream.
   *
   * @memberof Logcat
   */
  (0, _defineProperty2["default"])(this, "stop", function () {
    if (_this.stream) {
      _this.stream.cancel();
    }
    if (_this.timerID) {
      clearInterval(_this.timerID);
      _this.timerID = null;
    }
    _this.events.emit("end");
  });
  (0, _defineProperty2["default"])(this, "pollStream", function () {
    var self = _this;
    /* eslint-disable */
    var request = new proto.android.emulation.control.LogMessage();
    request.setStart(_this.offset);
    _this.emulator.getLogcat(request, {}, function (err, response) {
      if (err) {
        _this.stop();
      }
      if (response) {
        var nextOffset = response.getNext();
        if (nextOffset > self.offset) {
          self.offset = response.getNext();
          self.events.emit("data", response.getContents());
        }
      }
    });
  });
  // Uses streaming, this really locks up the ui, so best not to use for now.
  (0, _defineProperty2["default"])(this, "stream", function () {
    var self = _this;
    /* eslint-disable */
    var request = new proto.android.emulation.control.LogMessage();
    request.setStart(_this.offset);
    _this.stream = _this.emulator.streamLogcat(request);
    _this.stream.on("data", function (response) {
      self.offset = response.getNext();
      var contents = response.getContents();
      self.events.emit("data", contents);
    });
    _this.stream.on("error", function (error) {
      if (error.code = 1) {
        // Ignore we got cancelled.
      }
    });
  });
  /**
   * Requests the logcat stream, invoking the callback when a log line arrives.
   *
   * *Note:* Streaming can cause serious UI delays, so best not to use it.
   *
   * @param  {Callback} fnNotify when a new log line arrives.
   * @param  {number} refreshRate polling interval, or 0 if you wish to use streaming.
   * @memberof Logcat
   */
  (0, _defineProperty2["default"])(this, "start", function (fnNotify) {
    var refreshRate = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 1000;
    if (fnNotify) _this.on("data", fnNotify);
    _this.refreshRate = refreshRate;
    if (_this.refreshRate > 0) {
      _this.timerID = setInterval(function () {
        return _this.pollStream();
      }, _this.refreshRate);
    } else {
      _this.stream();
    }
    _this.events.emit("start");
  });
  if (uriOrEmulator instanceof _emulator_web_client.EmulatorControllerService) {
    this.emulator = uriOrEmulator;
  } else {
    this.emulator = new _emulator_web_client.EmulatorControllerService(uriOrEmulator, auth);
  }
  this.offset = 0;
  this.lastline = "";
  this.stream = null;
  this.events = new _events.EventEmitter();
  this.refreshRate = 1000;
  this.timerID = null;
});
var _default = Logcat;
exports["default"] = _default;
