// android-emulator-webrtc/dist/components/emulator/emulator.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
var _simple_png_view = _interopRequireDefault(require("./views/simple_png_view.js"));
var _webrtc_view = _interopRequireDefault(require("./views/webrtc_view.js"));
var _event_handler = _interopRequireDefault(require("./views/event_handler"));
var _jsep_protocol_driver = _interopRequireDefault(require("./net/jsep_protocol_driver.js"));
var Proto = _interopRequireWildcard(require("../../proto/emulator_controller_pb"));
var _emulator_web_client = require("../../proto/emulator_web_client");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } } /*
                                                                                                                                                                                                                                                                                                                                           * Copyright 2019 The Android Open Source Project
                                                                                                                                                                                                                                                                                                                                           *
                                                                                                                                                                                                                                                                                                                                           * Licensed under the Apache License, Version 2.0 (the "License")
                                                                                                                                                                                                                                                                                                                                           * you may not use this file except in compliance with the License.
                                                                                                                                                                                                                                                                                                                                           * You may obtain a copy of the License at
                                                                                                                                                                                                                                                                                                                                           *
                                                                                                                                                                                                                                                                                                                                           *     http://www.apache.org/licenses/LICENSE-2.0
                                                                                                                                                                                                                                                                                                                                           *
                                                                                                                                                                                                                                                                                                                                           * Unless required by applicable law or agreed to in writing, software
                                                                                                                                                                                                                                                                                                                                           * distributed under the License is distributed on an "AS IS" BASIS,
                                                                                                                                                                                                                                                                                                                                           * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
                                                                                                                                                                                                                                                                                                                                           * See the License for the specific language governing permissions and
                                                                                                                                                                                                                                                                                                                                           * limitations under the License.
                                                                                                                                                                                                                                                                                                                                           */
var PngView = (0, _event_handler["default"])(_simple_png_view["default"]);
var RtcView = (0, _event_handler["default"])(_webrtc_view["default"]);

/**
 * A React component that displays a remote android emulator.
 *
 * The emulator will mount a png or webrtc view component to display the current state
 * of the emulator. It will translate mouse events on this component and send them
 * to the actual emulator.
 *
 * #### Authentication Service
 *
 * The authentication service should implement the following methods:
 *
 * - `authHeader()` which must return a set of headers that should be send along with a request.
 * - `unauthorized()` a function that gets called when a 401 was received.
 *
 * #### Type of view
 *
 * You usually want this to be webrtc as this will make use of the efficient
 * webrtc implementation. The png view will request screenshots, which are
 * very slow, and require the envoy proxy. You should not use this for remote emulators.
 *
 * Note that chrome will not autoplay the video if it is not muted and no interaction
 * with the page has taken place. See https://developers.google.com/web/updates/2017/09/autoplay-policy-changes.
 *
 * #### Pressing hardware buttons
 *
 * This component has a method `sendKey` that sends a key to the emulator.
 * You can use this to send physical button events to the emulator for example:
 *
 * "AudioVolumeDown" - 	Decreases the audio volume.
 * "AudioVolumeUp"   -	Increases the audio volume.
 * "Power"	         -  The Power button or key, turn off the device.
 * "AppSwitch"       -  Should bring up the application switcher dialog.
 * "GoHome"          -  Go to the home screen.
 * "GoBack"          -  Open the previous screen you were looking at.
 *
 */
var Emulator = /*#__PURE__*/function (_Component) {
  (0, _inherits2["default"])(Emulator, _Component);
  var _super = _createSuper(Emulator);
  function Emulator(props) {
    var _this;
    (0, _classCallCheck2["default"])(this, Emulator);
    _this = _super.call(this, props);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "components", {
      webrtc: RtcView,
      png: PngView
    });
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "state", {
      audio: false
    });
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "componentDidMount", function () {
      _this.updateLocation();
    });
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "componentDidUpdate", function (prevProps) {
      if (prevProps.gps !== _this.props.gps) {
        _this.updateLocation();
      }
    });
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "updateLocation", function () {
      var gps = _this.props.gps;
      if (typeof gps === "undefined") {
        return;
      }
      var state = new Proto.GpsState();
      state.setLatitude(gps.latitude);
      state.setLongitude(gps.longitude);
      state.setAltitude(gps.altitude);
      state.setBearing(gps.heading);
      state.setSpeed(gps.speed);
      _this.emulator.setGps(state);
    });
    /**
     * Sends the given key to the emulator.
     *
     * You can use this to send physical hardware events to the emulator for example:
     *
     * "AudioVolumeDown" - 	Decreases the audio volume.
     * "AudioVolumeUp"   -	Increases the audio volume.
     * "Power"	         -  The Power button or key, turn off the device.
     * "AppSwitch"       -  Should bring up the application switcher dialog.
     * "GoHome"          -  Go to the home screen.
     * "GoBack"          -  Open the previous screen you were looking at.
     *
     * See https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key/Key_Values for
     * a list of valid values you can send as well.
     */
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "sendKey", function (key) {
      var request = new Proto.KeyboardEvent();
      request.setEventtype(Proto.KeyboardEvent.KeyEventType.KEYPRESS);
      request.setKey(key);
      _this.jsep.send("keyboard", request);
    });
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "_onAudioStateChange", function (s) {
      var onAudioStateChange = _this.props.onAudioStateChange;
      _this.setState({
        audio: s
      }, function () {
        onAudioStateChange(s);
      });
    });
    var uri = props.uri,
      auth = props.auth,
      poll = props.poll,
      onError = props.onError;
    _this.emulator = new _emulator_web_client.EmulatorControllerService(uri, auth, onError);
    _this.rtc = new _emulator_web_client.RtcService(uri, auth, onError);
    _this.jsep = new _jsep_protocol_driver["default"](_this.emulator, _this.rtc, poll);
    _this.view = /*#__PURE__*/_react["default"].createRef();
    return _this;
  }
  (0, _createClass2["default"])(Emulator, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
        width = _this$props.width,
        height = _this$props.height,
        view = _this$props.view,
        poll = _this$props.poll,
        muted = _this$props.muted,
        onStateChange = _this$props.onStateChange,
        onError = _this$props.onError,
        volume = _this$props.volume;
      var SpecificView = this.components[view] || RtcView;
      console.log("render ".concat(width, "x").concat(height));
      return /*#__PURE__*/_react["default"].createElement(SpecificView, {
        ref: this.view,
        width: width,
        height: height,
        emulator: this.emulator,
        jsep: this.jsep,
        onStateChange: onStateChange,
        poll: poll,
        muted: muted,
        volume: volume,
        onError: onError,
        onAudioStateChange: this._onAudioStateChange
      });
    }
  }], [{
    key: "getDerivedStateFromProps",
    value: function getDerivedStateFromProps(nextProps, prevState) {
      if (nextProps.view === "png") return {
        audio: false
      };
      return prevState;
    }
  }]);
  return Emulator;
}(_react.Component);
(0, _defineProperty2["default"])(Emulator, "propTypes", {
  /** gRPC Endpoint where we can reach the emulator. */
  uri: _propTypes["default"].string.isRequired,
  /** The authentication service to use, or null for no authentication. */
  auth: _propTypes["default"].object,
  /** True if the audio should be disabled. This is only relevant when using the webrtc engine. */
  muted: _propTypes["default"].bool,
  /** Volume between [0, 1] when audio is enabled. 0 is muted, 1.0 is 100% */
  volume: _propTypes["default"].number,
  /** Called upon state change, one of ["connecting", "connected", "disconnected"] */
  onStateChange: _propTypes["default"].func,
  /** Called when the audio becomes (un)available. True if audio is available, false otherwise. */
  onAudioStateChange: _propTypes["default"].func,
  /** The width of the component */
  width: _propTypes["default"].number,
  /** The height of the component */
  height: _propTypes["default"].number,
  /** The underlying view used to display the emulator, one of ["webrtc", "png"] */
  view: _propTypes["default"].oneOf(["webrtc", "png"]).isRequired,
  /** A [GeolocationCoordinates](https://developer.mozilla.org/en-US/docs/Web/API/GeolocationCoordinates) like object indicating where the device is. */
  gps: _propTypes["default"].object,
  /** True if polling should be used, only set this to true if you are using the go webgrpc proxy. */
  poll: _propTypes["default"].bool,
  /** Callback that will be invoked in case of gRPC errors. */
  onError: _propTypes["default"].func
});
(0, _defineProperty2["default"])(Emulator, "defaultProps", {
  view: "webrtc",
  auth: null,
  poll: false,
  muted: true,
  volume: 1.0,
  onError: function onError(e) {
    console.error(e);
  },
  onAudioStateChange: function onAudioStateChange(s) {
    console.debug("emulator audio: " + s);
  },
  onStateChange: function onStateChange(s) {
    console.debug("emulator state: " + s);
  }
});
var _default = Emulator;
exports["default"] = _default;
