// android-emulator-webrtc/dist/components/emulator/views/simple_png_view.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireWildcard(require("react"));
var Proto = _interopRequireWildcard(require("../../../proto/emulator_controller_pb"));
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
var EmulatorPngView = function EmulatorPngView(_ref) {
  var emulator = _ref.emulator,
    onStateChange = _ref.onStateChange,
    poll = _ref.poll,
    width = _ref.width,
    height = _ref.height,
    dataTestId = _ref["data-testid"];
  var _useState = (0, _react.useState)(""),
    _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
    png = _useState2[0],
    setPng = _useState2[1];
  var _useState3 = (0, _react.useState)("connecting"),
    _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
    connect = _useState4[0],
    setConnect = _useState4[1];
  var screenShot = null;
  (0, _react.useEffect)(function () {
    startStream();
    return function () {
      setConnect("disconnected");
      if (screenShot) {
        screenShot.cancel();
      }
    };
  }, [width, height]);
  (0, _react.useEffect)(function () {
    if (onStateChange) {
      onStateChange(connect);
    }
  }, [connect]);
  var startStream = function startStream() {
    setConnect("connecting");
    var request = new Proto.ImageFormat();
    if (!isNaN(width)) {
      request.setWidth(Math.floor(width));
    }
    if (!isNaN(height)) {
      request.setWidth(Math.floor(height));
    }
    if (poll && connect !== "disconnected") {
      emulator.getScreenshot(request, {}, function (err, response) {
        setConnect("connected");
        setPng("data:image/jpeg;base64," + response.getImage_asB64());
        startStream();
      });
    } else {
      var receivedImage = false;
      screenShot = emulator.streamScreenshot(request);
      screenShot.on("data", function (response) {
        receivedImage = true;
        setConnect("connected");
        setPng("data:image/jpeg;base64," + response.getImage_asB64());
      });
      screenShot.on("error", function (e) {
        console.warn("Screenshot stream broken", e);
        if (receivedImage) {
          setConnect("connecting");
          startStream();
        } else {
          setConnect("disconnected");
        }
      });
    }
  };
  var preventDragHandler = function preventDragHandler(e) {
    e.preventDefault();
  };
  return /*#__PURE__*/_react["default"].createElement("div", {
    width: width,
    style: {
      display: "block",
      position: "relative",
      height: "100%",
      objectFit: "contain",
      objectPosition: "center"
    },
    onDragStart: preventDragHandler
  }, /*#__PURE__*/_react["default"].createElement("img", {
    src: png,
    width: "100%",
    draggable: "false",
    style: {
      pointerEvents: "none"
    }
  }));
};
EmulatorPngView.propTypes = {
  emulator: _propTypes["default"].object,
  onStateChange: _propTypes["default"].func,
  poll: _propTypes["default"].bool,
  width: _propTypes["default"].number,
  height: _propTypes["default"].number,
  "data-testid": _propTypes["default"].string
};
var _default = EmulatorPngView;
exports["default"] = _default;
